dlt/destinations/impl/filesystem/sql_client.py

```diff
@@ -0,0 +1,123 @@
+from typing import Any, Iterator, AnyStr, List, cast
+
+import os
+
+import duckdb
+
+import sqlglot
+import sqlglot.expressions as exp
+
+from contextlib import contextmanager
+
+from dlt.common.destination.reference import DBApiCursor
+from dlt.common.destination.typing import PreparedTableSchema
+
+from dlt.destinations.sql_client import raise_database_error
+from dlt.destinations.fs_client import FSClientBase
+
+from dlt.destinations.impl.duckdb.sql_client import DuckDbSqlClient
+from dlt.destinations.impl.duckdb.factory import duckdb as duckdb_factory
+
+SUPPORTED_PROTOCOLS = ["gs", "gcs", "s3", "file", "memory"]
+
+
+class FilesystemSqlClient(DuckDbSqlClient):
+    def __init__(self, fs_client: FSClientBase, protocol: str, dataset_name: str) -> None:
+        super().__init__(
+            dataset_name=dataset_name,
+            staging_dataset_name=None,
+            credentials=None,
+            capabilities=duckdb_factory()._raw_capabilities(),
+        )
+        self.fs_client = fs_client
+        self.existing_views: List[str] = []  # remember which views already where created
+        self.protocol = protocol
+        self.is_local_filesystem = protocol == "file"
+
+        if protocol not in SUPPORTED_PROTOCOLS:
+            raise NotImplementedError(
+                f"Protocol {protocol} currently not supported for FilesystemSqlClient. Supported"
+                f" protocols are {SUPPORTED_PROTOCOLS}."
+            )
+
+        # set up duckdb instance
+        self._conn = duckdb.connect(":memory:")
```

Comment:
I did this now, it will retain the in memory version of the database and fall back to the mechanism in the parent class when using an externally passed in database connection